### DISCUSSION TITLE
BTCPay Shop / Hosting Catalog

### DISCUSSION THREAD

**Discussion by onleines** [btcpayserver/btcpayserver#1817] [Ideas / Feature Requests]
It would be desirable if there would be an app within the BTCPay server which could be used as an online shop light (BTCPay Shop).

If you only want to sell a few products against Bitcoin (Only) on your website, you always have to do this together with an existing online shop software like WooCommerce.

The idea is to provide a simple online shop so that a few physical products can be sold on your own website without having to integrate an external shop solution.

If a merchant wants to sell digital products instead of physical products, this can be done with the help of the POS app. The buyer only needs to enter his email address to receive the digital goods. 

Just as it was implemented in the BTCPay sample application with the event ticket.

Perhaps the existing POS app can be extended to include the shipping address details?
Maybe the shipping costs can still be a challenge. 

Bitpay has implemented a comparable solution with the Hosting Catalog as follows:

![1_hosted_catalog](https://user-images.githubusercontent.com/22472943/89460051-f6c70f00-d769-11ea-8e61-56373eabf085.jpg)

![2_Hosted_Catalog](https://user-images.githubusercontent.com/22472943/89460106-09414880-d76a-11ea-86eb-be4d72471a4d.jpg)

Add a new item

![3_Hosted_Catalog_Item](https://user-images.githubusercontent.com/22472943/89460150-1c541880-d76a-11ea-8702-391a16c87579.jpg)

![4_Hosted_Catalog_Item](https://user-images.githubusercontent.com/22472943/89460172-2544ea00-d76a-11ea-91eb-d8bc110af602.jpg)

This is what the shop looks like for the customers:

[https://bitpay.com/cart?id=6fxiY3ywaBbsWq5TtTyfqo](https://bitpay.com/cart?id=6fxiY3ywaBbsWq5TtTyfqo)

 **Bitpay Help for the Hosted Catalog** 

BitPay will host a shopping cart to process sales and payments for your catalog.

This page explains the detailed instructions for adding your items to the BitPay database, and how to let your customers order these items using bitcoins.

Step 1. Create a Hosted Checkout Button
The item you create with the Catalog Item generator will store all of the embedded data on BitPay's servers.

You can edit the item data at any time on our site through the  My Catalog Items link.

You can make a different Catalog Item for each product you want to sell.

Place the checkout button HTML code on the catalog pages of your website, or print the QR code on any poster, ad, or sign.

Step 2. Buyer Orders from your Website
When a buyer clicks the website button or scans the QR image, they land on a BitPay shopping cart with this item added. If they already have an open shopping cart for you, this item will be added to the cart.

Shoppers can only have items from one merchant in a cart.

Shoppers can change the quantity they want to order, or return to your website and add more items.

If you update the price or description of an item at BitPay, it will update any open shopping carts that contain that item.

Before proceeding to checkout, the Shopper must enter the fields you selected to be required for the items (Name, Address, Email, etc).

Step 3. Buyer Pays for the Order
When payment is received and confirmed (according to your  Risk/Speed settings), BitPay will send an email to both the buyer and the merchant with a detailed receipt and summary of the order.

The buyer will also be presented with a receipt in their browser.

**Comment by Zaxounette** [comment]
Related to this discussion, the following code was made public today:
https://gist.github.com/j-chimienti/9453454ed570bd5665011459a48aac7c

It's an example of an implementation of a shipping info popup on top of the current POS.

**Comment by NicolasDorier** [comment]
@onleines how is it different from the PoS app in btcpay?

**Comment by crypto-credit** [comment]
Just set up my first btcpay account and would like to add a couple of comments:
The POS app works well and looks nice as a stand alone store but i noticed that i could only add the email field to the payment. 
The lightning invoice has extra fields under buyer information that cannot be utilized with the existing set up.
The option to customise field description under buyer info would be useful.
I also agree that it would be good to be able to use btcpay without having to use WooCommerce.

**Comment by onleines** [comment]
@NicolasDorier 

I would say (almost) no different. You can already use the POS app to sell digital products that are sent to an email address.

There is already a great example with the Event Ticket, which is sent by email.

https://github.com/btcpayserver/btcTransmuter/blob/master/docs/EmailReceiptsPreset.md

I have rebuilt it and described it here:

https://coincharge.io/en/portfolio/sell-digital-goods-with-bitcoin-and-deliver-them-by-email/

In addition to the email address, the postal address would have to be requested in order to send the physical goods.

Theoretically only the additional address fields are needed then, right?

Maybe the shipping costs as well. Bitpay does it with a fixed amount.  But here you should offer several regions as an option. As a European merchant I have different costs for domestic, European and the rest of the world.

In the POS App there is already a comparable function with the tip function.

I believe that the most important functions are already available in the POS App, such as product display and shopping cart function.

Unfortunately I am not a programmer and therefore I can only judge it to a limited extent.

**Comment by pavlenex** [comment]
This issue was posted at the same time @rockstardev and I were having a convo about it, speaking of timing.

I believe instead of cluttering PoS with options we should develop a new app that is based quite a lot on it, but focused on a specific use-case - simple online store. You see if we keep adding features to a specific app, unfortunatelly, people will not know about it. The e-comm useage would be hidden away in docs. However, if we re-use something from PoS and modify it to specifically suit the need of a very simple e-com merchant, I think people would use it more.

I have some ideas and following up on discussion here. As soon as I free up some time I'll do a wireframe.

**Comment by pavlenex** [comment]
Closing this since using BTCPay Forms is possible with POS https://docs.btcpayserver.org/Forms/#setting-up-your-stores-first-custom-form and allows you to collect customer address.
